collect_app/src/test/java/org/odk/collect/android/preferences/AdminPreferencesActivityTest.java

```diff
@@ -0,0 +1,126 @@
+package org.odk.collect.android.preferences;
+
+import android.content.SharedPreferences;
+import android.preference.CheckBoxPreference;
+
+import org.hamcrest.Description;
+import org.hamcrest.StringDescription;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.odk.collect.android.BuildConfig;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+import static org.robolectric.util.FragmentTestUtil.startFragment;
+
+/**
+ * Unit test for checking {@link SharedPreferences}'s behaviour
+ */
+@Config(constants = BuildConfig.class)
+@RunWith(RobolectricTestRunner.class)
+public class AdminPreferencesActivityTest {
+
+    private SortedMap<String, String> preferenceHashMap;
```

Comment:
A hash map isn't actually sorted so this name is misleading! I would call it something like `preferenceIDsToAdminKeys`.